cognite/powerops/resync/models/_shared_v1_v2/production_model.py

```diff
@@ -86,25 +92,47 @@ def _plant_to_inlet_reservoir_with_losses(
     """
 
     def get_connection_path_from_last_visited(visited_paths: list, last_visited_id: int) -> list:
+        """Return the correct sequence of visited connections based on the last visited connection among
+        the list of connection paths visited
+
+            Parameters
+            ----------
+            visited_paths : list
+                 A list that holds the lists of each visited connection path. A connection path will be a list of
+                 connection IDs, e.g. [1,2,3] means connection with ID 1 was first visited, then 2, then 3
+            last_visited_id: int
+                The ID of the connection that was last visited. This will be the last item in one of the lists of
+                visited paths
+
+            Returns
+            -------
+            list
+                The path or sequence of connections that has the last_visited_id as its last visited connection among
+                the visited_paths
+        """
         for connection in visited_paths:
             if connection[-1] == last_visited_id:
                 connection_path_index = visited_paths.index(connection)
                 return visited_paths[connection_path_index]
 
-    def calculate_losses_from_conncetion_path(all_junctions, all_tunnels, connection_by_id, connection_path):
+    def calculate_losses_from_connection_path(
+        all_junctions: dict, all_tunnels: dict, connection_by_id: int, connection_path: list[int]
+    ):
+        """Loop through connections in connection path, retrieve the losses for that connection among the all_juntions
+        or all_tunnels based on the type of connection, and sum up the total losses from the connection path
+        """
         sum_losses = 0
         order_to_loss_factor_key = {0: "loss_factor_1", 1: "loss_factor_2"}
         for connection_id in connection_path:
             connection = connection_by_id[connection_id]
-            if connection.get("to_type") == "junction":
+            connection_name = connection["to"]
+            if connection_name in all_junctions:
                 if connection.get("order") in order_to_loss_factor_key:
-                    junction_name = connection["to"]
-                    junction_losses = all_junctions[junction_name]
+                    junction_losses = all_junctions[connection_name]
                     loss_order = connection["order"]
                     sum_losses += junction_losses[order_to_loss_factor_key[loss_order]]
-            elif connection.get("to_type") == "tunnel":
-                tunnel_name = connection["to"]
-                tunnel_loss = all_tunnels[tunnel_name]["loss_factor"]
+            elif connection_name in all_tunnels:
+                tunnel_loss = all_tunnels[connection_name]["loss_factor"]
```

Comment:
Nice! This seems like a "safer" way to do it, although it might be a bit slower 👍